internal/kafka/internal/services/kafka_test.go

```diff
@@ -2519,20 +2530,63 @@ func Test_kafkaService_DeprovisionExpiredKafkas(t *testing.T) {
 			},
 			wantErr: true,
 			setupFn: func() {
-				mocket.Catcher.Reset().NewMock().WithQuery(`SELECT * FROM "kafka_requests" WHERE instance_type IN ($1) AND status NOT IN ($2,$3)`).WithReply([]map[string]interface{}{{"id": "kafkainstance1", "instance_type": instanceType, "size_id": instanceSize}})
+				mocket.Catcher.Reset().NewMock().WithQuery(`SELECT * FROM "kafka_requests" WHERE status NOT IN ($2,$3) AND expires_at IS NOT NULL`).WithReply([]map[string]interface{}{{"id": "kafkainstance1", "instance_type": instanceType, "size_id": instanceSize}})
 				mocket.Catcher.NewMock().WithQuery(`UPDATE "kafka_requests" SET "status"=$1,"updated_at"=$2 WHERE id IN ($3)`).WithError(fmt.Errorf("an update error"))
 				mocket.Catcher.NewMock().WithExecException().WithQueryException()
 			},
 		},
 		{
-			name: "success when database does not throw an error",
+			name: "when a kafka instance has expired it marks it as deprovisioned successfully",
 			fields: fields{
 				connectionFactory: db.NewMockConnectionFactory(nil),
+				testDBNowFunc: func() time.Time {
+					return dbTime
+				},
 			},
 			wantErr: false,
 			setupFn: func() {
-				mocket.Catcher.Reset().NewMock().WithQuery(`SELECT * FROM "kafka_requests" WHERE instance_type IN ($1) AND status NOT IN ($2,$3)`).WithReply([]map[string]interface{}{{"id": "kafkainstance1", "instance_type": instanceType, "size_id": instanceSize}})
-				mocket.Catcher.NewMock().WithQuery(`UPDATE "kafka_requests" SET "status"=$1,"updated_at"=$2 WHERE id IN ($3)`)
+				mocket.Catcher.Reset().NewMock().
+					WithQuery(`SELECT * FROM "kafka_requests" WHERE status NOT IN ($1,$2) AND expires_at IS NOT NULL`).
+					WithReply([]map[string]interface{}{{"id": "kafkainstance1", "instance_type": instanceType, "size_id": instanceSize, "expires_at": &expiredTime}})
```

Comment:
I believe using `any` instead of `interface{}` would be more readable (`any` is just an alias for `interface{}`)

Reply:
Not going to change it to be consistent with the other elements in the file. At some point if we conclude that `any` is preferred we can change sets of files as a general practice throughout the project. Maybe a good idea however would be to reference the same type as the one defined in the signatures of the library being used in the case of 3rd party libraries to have a 1:1 visual mapping. In any case not something to be dealt with in this PR